salt/utils/vmware.py

```diff
@@ -187,71 +303,61 @@ def get_service_instance(host, username, password, protocol=None, port=None):
     service_instance = GetSi()
     if service_instance:
         if service_instance._GetStub().host == ':'.join([host, str(port)]):
-            return service_instance
+            log.trace('Using cached service instance')
+            if salt.utils.is_proxy():
```

Comment:
Removed as part of 14ea29f446563c1c8db255e0a458eaaeeaac5c67, but not sure it works